src/ring_theory/ideal/over.lean

```diff
@@ -223,6 +232,17 @@ begin
     simpa [← ring_hom.ker_eq_comap_bot] using hIP},
 end
 
+/-- `comap (algebra_map R S)` is a surjection from the max spec of `S` to max spec of `R`.
+`hP : (algebra_map R S).ker ≤ P` is a slight generalization of the extension being injective -/
+lemma exists_ideal_over_maximal_of_is_integral (H : algebra.is_integral R S)
+  (P : ideal R) [P_max : is_maximal P] (hP : (algebra_map R S).ker ≤ P) :
+  ∃ (Q : ideal S), is_maximal Q ∧ Q.comap (algebra_map R S) = P :=
+begin
+  obtain ⟨Q, ⟨Q_prime, hQ⟩⟩ := exists_ideal_over_prime_of_is_integral' H P hP,
+  haveI : Q.is_prime := Q_prime,
+  refine ⟨Q, ⟨is_maximal_of_is_integral_of_is_maximal_comap H _ (hQ.symm ▸ P_max), hQ⟩⟩,
```

Comment:
Constructor brackets associate to the right, so `⟨Q, Q_prime, hQ⟩` is the same as `⟨Q, ⟨Q_prime, hQ⟩⟩`. Since it's clear what we are (de)structuring (`∃ _, _ ∧ _`), both are equally readable to me and so I prefer the one with less symbols.
```suggestion
  obtain ⟨Q, Q_prime, hQ⟩ := exists_ideal_over_prime_of_is_integral' H P hP,
  haveI : Q.is_prime := Q_prime,
  exact ⟨Q, is_maximal_of_is_integral_of_is_maximal_comap H _ (hQ.symm ▸ P_max), hQ⟩,
```